src/pymorize/core/cmorizer.py

```diff
@@ -364,7 +364,30 @@ def _rules_expand_drvs(self):
             else:
                 cloned_rules = rule.expand_drvs()
                 for rule in cloned_rules:
-                    new_rules.append(rule)
+                    # Rule has a table_id or a table_name, so it should only
+                    # match that table
+                    if hasattr(rule, "table_id"):
+                        if isinstance(rule.table_id, str):
+                            rule.table_id = [
+                                rule.table_id,
+                            ]
+                        logger.info(f"Specified table_id as {rule.table_id=}")
+                        for drv in rule.data_request_variables:
+                            if drv.table_header.table_id in rule.table_id:
+                                logger.info(f"Adding rule/table combo for {drv}")
+                                new_rules.append(rule)
+                    elif hasattr(rule, "table_name"):
+                        if isinstance(rule.table_name, str):
+                            rule.table_name = [
+                                rule.table_name,
+                            ]
+                        logger.info(f"Specified table_name as {rule.table_name=}")
+                        for drv in rule.data_request_variables:
+                            if drv.table_header.table_id in rule.table_name:
+                                logger.info(f"Adding rule/table combo for {drv}")
+                                new_rules.append(rule)
+                    else:
+                        new_rules.append(rule)
```

Comment:
Let me see if I get it. What this is currently doing is:
1. expand one rule into multiple rules with `rule.expand_drvs()` and return clones of them, one per each table found in the CVs
2. Include only the rules matching the tables specified by the user

In essence we are cloning and then only using part of it. I am guessing that is not a problem for performance right? What we are cloning is not the data but rather rule objects, so that should not be a problem with performance right?

The other option would be to include this logic directly in `rule.expand_drvs` so that rules are only copied the minimum number of times. That would make the code a bit less readable, so as long as you estimate that this would never be bad for performance the code can stay as it is.

Reply:
I don't think this will be a performance choke point. It would be much cleaner to put this expand_drvs directly on the rule (after all, it's the rule's job to know how many copies of itself it needs to make, not really the cmorizers), but in the interest of laziness, I would leave it here for now...probably I will regret that later.